cms/djangoapps/api/v1/serializers/course_runs.py

```diff
@@ -0,0 +1,165 @@
+            return instance
+
+    def update_team(self, instance, team):
+        CourseAccessRole.objects.filter(course_id=instance.id).delete()
+
+        # TODO In the future we can optimize by getting users in a single query.
+        CourseAccessRole.objects.bulk_create([
+            CourseAccessRole(course_id=instance.id, role=member['role'], user=User.objects.get(username=member['user']))
+            for member in team
+        ])
+
+
+class CourseRunCreateSerializer(CourseRunSerializer):
+    org = serializers.CharField(source='id.org')
+    number = serializers.CharField(source='id.course')
+    run = serializers.CharField(source='id.run')
+
+    def create(self, validated_data):
+        _id = validated_data.pop('id')
+        team = validated_data.pop('team', [])
+        user = self.context['request'].user
+
+        with transaction.atomic():
+            instance = create_new_course(user, _id['org'], _id['course'], _id['run'], validated_data)
+            self.update_team(instance, team)
+            return instance
+
+
+class CourseRunImageField(serializers.ImageField):
+    def get_attribute(self, instance):
+        return course_image_url(instance)
+
+    def run_validation(self, data=empty):
+        content_type = getattr(data, 'content_type')
+        if content_type not in IMAGE_TYPES:
+            raise serializers.ValidationError('{} is not a supported image type'.format(content_type))
+
+        return super(CourseRunImageField, self).run_validation(data)
+
+    def to_representation(self, value):
+        # Value will always be the URL path of the image.
+        request = self.context['request']
+        return request.build_absolute_uri(value)
+
+
+class CourseRunImageSerializer(serializers.Serializer):
+    # We set an empty default to prevent the parent serializer from attempting
+    # to save this value to the Course object.
+    card_image = CourseRunImageField(source='course_image', default=empty)
+
+    def update(self, instance, validated_data):
+        course_image = validated_data['course_image']
+        course_image.name = 'course_image.' + IMAGE_TYPES[course_image.content_type]
+        update_course_run_asset(instance.id, course_image)
+
+        instance.course_image = course_image.name
+        modulestore().update_item(instance, self.context['request'].user.id)
+        return instance
+
+
+class CourseRunRerunSerializer(CourseRunTeamSerializerMixin, serializers.Serializer):
+    title = serializers.CharField(source='display_name', required=False)
+    run = serializers.CharField(source='id.run')
+    schedule = CourseRunScheduleSerializer(source='*', required=False)
+
+    def validate_run(self, value):
+        course_run_key = self.instance.id
+        store = modulestore()
+        with store.default_store('split'):
+            new_course_run_key = store.make_course_key(course_run_key.org, course_run_key.course, value)
+        if store.has_course(new_course_run_key, ignore_case=True):
```

Comment:
Shouldn't this line be indented as part of the `with` code block?

Reply:
I don't know. Much of the modulestore code here was pulled from elsewhere. This particular block bit of code was taken from https://github.com/edx/edx-platform/blob/d0cfa7fa5a73e3491ff83c9e4def5db88a30b5ec/cms/djangoapps/contentstore/views/course.py#L874-L881.